cmd/build/circleciv2.go

```diff
@@ -0,0 +1,80 @@
+package build
+
+import "fmt"
+
+//CircleCIv2 represents a circle ci v2 build provider
+type CircleCIv2 struct{}
+
+//ProvideArtifacts is the Provider implementation
+func (provider CircleCIv2) ProvideArtifacts(context Context) ([]*Artifact, error) {
+
+	//output the following:
+	//circleci/config.yml
+	//circleci/config.env
+
+	contextTemplate := getContextTemplate(context)
+
+	artifacts := []*Artifact{}
+	artifacts = append(artifacts, createArtifact(".circleci/config.yml", getCircleCIv2YAML()))
+	artifacts = append(artifacts, createArtifact(".circleci/config.env", getConfigEnv(contextTemplate)))
+
+  fmt.Println()
+	fmt.Println(`Be sure to supply the following environment variables in your Circle CI build:
+  AWS_ACCESS_KEY_ID (terraform state show aws_iam_access_key.cicd_keys)
+  AWS_SECRET_ACCESS_KEY (terraform state show aws_iam_access_key.cicd_keys)
+  AWS_DEFAULT_REGION=us-east-1`)
+	fmt.Println()
+
+	return artifacts, nil
+}
+
+func getCircleCIv2YAML() string {
+	template := `
+version: 2
+jobs:
+  build:
+    docker:
+      - image: quay.io/turner/fargate-cicd
+    environment:
+      VAR: .circleci/config.env
+    steps:
+      - checkout
+      - setup_remote_docker:
+          version: 18.06.0-ce
+      - run:
+          name: Set docker image
+          command: |
+            source ${VAR}
```

Comment:
In literal shell code, variable expansions should normally be double-quoted in case they contain weird characters, so `$VAR` should have double-quotes around it (here and in the rest of this mini-script). The right hand side of an assignment is exempt from this requirement; you only need quotes there if you have _literal_ weird characters, usually spaces. So the various `$VAR`s are the only expansions calling for quotes in this snippet.

As a side note, the curly braces (in `${VAR}`) are optional unless you're doing a special expansion (substring extraction, array indexing, prefix or suffix omission...) or the next character looks like part of a shell identifier (`"${VAR}_$OTHERVAR"`). There's certainly an argument for the consistency of using them all the time, but I prefer to leave them off when not required, especially since they might briefly confuse folks scanning a Go template for `{{`..`}}` actions.